p2p/shard_syncer.py

```diff
@@ -99,7 +99,7 @@ def propose(self) -> Collation:
         self.shard.add_collation(collation)
 
         # broadcast collation
-        for peer in self.peer_pool.peers:
+        for peer in self.peer_pool._peers:
```

Comment:
We shouldn't be accessing private properties.  Maybe we should implement `PeerPool.__iter__` so that we can do `for peer in peer_pool`

Reply:
So, the reason why I made that private is because I want to have, eventually, all PeerPool subscribers using one of the APIs that select peers based on a given criteria instead of going through all connected peers, but maybe there are valid use cases for accessing all connected peers, like here, where we want to broadcast to all... 

I'd prefer to keep the `.peers` property, though, instead of implementing `__iter__`, but can't really think of a good reason for going one way or the other, so happy to do `__iter__` if you feel strong about it

Reply:
Don't feel strongly.  `PeerPool.peers` is fine with me.

Reply:
Ok, I've reverted it back to `.peers`; ready for another look